### PR TITLE
use new blob lib in func tests 

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

upgrade func tests,
in next PR I will move test configs to config class.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java [hunk 1]
```diff
@@ -96,11 +93,11 @@ public void should_throw_exception_when_requested_service_is_not_configured() th
...
         String testContainerName = conf.getString("test-storage-container-name");
         String sasToken = testHelper.getSasToken(testContainerName, this.testUrl);
-        CloudBlobContainer testSasContainer =
-            testHelper.getCloudContainer(sasToken, testContainerName, this.blobContainerUrl);
+        BlobContainerClient testSasContainer =
+            testHelper.getContainerClient(sasToken, testContainerName, this.blobContainerUrl);
 
         destZipFilename = testHelper.getRandomFilename();
-        testHelper.uploadAndLeaseZipFile(
+        this.leaseId = testHelper.uploadAndLeaseZipFile(
```
Comment (mdayican) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java:100]: we are uploading and after that we are leasing it, we are using random files names.
do we need to lease ?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java [hunk 2]
```diff
@@ -50,44 +51,41 @@ public void setUp() throws Exception {
...
-            );
+        String connectionString = String.format(
+            "DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s",
+            config.getString("test-storage-account-url"),
+            config.getString("test-storage-account-name"),
+            config.getString("test-storage-account-key")
+        );
+        leaseClientProvider =  blobClient -> new BlobLeaseClientBuilder().blobClient(blobClient).buildClient();
+        BlobServiceClientBuilder blobServiceClientBuilder = new BlobServiceClientBuilder()
+            .connectionString(connectionString);
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java:62]: we can still use `StorageSharedKeyCredential` here and set `url` with `builder.endpoint(url)`

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java [hunk 3]
```diff
@@ -103,11 +103,12 @@ private void times(int times, Runnable actionToRun) {
...
-
-        return stream(blobs.spliterator(), false)
-            .filter(b -> b.getUri().getPath().contains(fileName))
-            .collect(toList());
+    private List<BlobItem> searchByName(BlobContainerClient container, String fileName) {
+        ListBlobsOptions listOptions = new ListBlobsOptions();
+        listOptions.getDetails().setRetrieveSnapshots(true);
+        return    container.listBlobs(listOptions,null, null)
+            .stream()
+                .filter(b -> b.getName().equals(fileName))
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:111]: maybe we can try out this feature?:

```java
/**
 * Filters the results to return only blobs whose names begin with the specified prefix. May be null to return all
 * blobs.
 *
 * @param prefix A prefix that a blob must match to be returned
 * @return the updated ListBlobsOptions object
 */
public ListBlobsOptions setPrefix(String prefix) {
    this.prefix = prefix;
    return this;
}
```

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java [hunk 4]
```diff
@@ -103,11 +103,12 @@ private void times(int times, Runnable actionToRun) {
...
+                .collect(toList());
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:112]: what is happening with the whitespace overhere :see_no_evil:

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java (hunk 1 above)
Comment (doncem) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java:100]: i guess not :)

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java (hunk 4 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:112]: see_no_evil:

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java (hunk 3 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:111]: let's try

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java (hunk 1 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/GetSasTokenTest.java:100]: I will move later.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java [hunk 5]
```diff
@@ -50,44 +52,43 @@ public void setUp() throws Exception {
         this.isProxyEnabled = Boolean.valueOf(config.getString("proxyout.enabled"));
         this.fluxFuncTest = config.getBoolean("flux-func-test");
 
-        StorageCredentialsAccountAndKey storageCredentials =
-            new StorageCredentialsAccountAndKey(
+        leaseClientProvider =  blobClient -> new BlobLeaseClientBuilder().blobClient(blobClient).buildClient();
+
+        StorageSharedKeyCredential storageCredentials =
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java:57]: we can simplify with just `var`. and though avoid that new line. just personally buggers if it's new line :)

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java [hunk 6]
```diff
@@ -50,44 +52,43 @@ public void setUp() throws Exception {
...
+            new StorageSharedKeyCredential(
                 config.getString("test-storage-account-name"),
                 config.getString("test-storage-account-key")
             );
 
+        BlobServiceClientBuilder blobServiceClientBuilder = new BlobServiceClientBuilder()
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java:63]: `var` here too?

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java [hunk 7]
```diff
@@ -103,11 +103,12 @@ private void times(int times, Runnable actionToRun) {
         IntStream.range(0, times).forEach(index -> actionToRun.run());
     }
 
-    private List<ListBlobItem> searchByName(CloudBlobContainer container, String fileName) {
-        Iterable<ListBlobItem> blobs = container.listBlobs(null, true, EnumSet.of(SNAPSHOTS), null, null);
-
-        return stream(blobs.spliterator(), false)
-            .filter(b -> b.getUri().getPath().contains(fileName))
+    private List<BlobItem> searchByName(BlobContainerClient container, String fileName) {
+        ListBlobsOptions listOptions = new ListBlobsOptions();
```
Comment (doncem) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:107]: `var`? just maybe in general let's go through the pr and add `var`s where makes sense

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java (hunk 6 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BaseFunctionalTest.java:63]: next pr.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java (hunk 7 above)
Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java:107]: next pr.

**Pull request merged**